packages/appium/docs/en/ecosystem/build-drivers.md

```diff
@@ -2,4 +2,617 @@
+one!
+
+## Basic requirements for Appium drivers
+
+These are the things your driver *must* do (or be), if you want it to be a valid Appium driver.
+
+### Node.js package with an Appium manifest
+
+All Appium drivers are fundamentally Node.js packages, and therefore must have a valid
+`package.json`. Your driver is not limited to Node.js, but whatever you do with your driver must
+ultimately be triggered by Node.js.
+
+Your `package.json` must include `appium` as a `peerDependency`. The requirements for the
+dependency versions should be as loose as possible (unless you happen to know your driver will only
+work with certain versions of Appium). For Appium 2.0, for example, this would look something like
+`^2.0.0`, declaring that your driver works with any version of Appium that starts with 2.x.
+
+Your `package.json` must contain an `appium` field, like this (we call this the 'Appium manifest'):
+
+    ```json
+    {
+        ...,
+        "appium": {
+            "driverName": "fake",
+            "automationName": "Fake",
+            "platformNames": [
+              "Fake"
+            ],
+            "mainClass": "FakeDriver"
+        },
+        ...
+    }
+    ```
+
+The required subfields are:
+
+* `driverName`: this should be a short name for your driver.
+* `automationName`: this should be the string users will use for their `appium:automationName`
+capability to tell Appium to use *your* driver.
+* `platformNames`: this is an array of platform names considered valid for your driver. When a user
+sends in the `platformName` capability to start a session, it must be included in this list for
+your driver to handle the session.
+* `mainClass`: this is a named export from your `main` field. It must be a class which extends
+Appium's `BaseDriver` (see below).
+
+### Extend Appium's `BaseDriver` class
+
+Ultimately, your driver is much easier to write because most of the hard work of implementing the
+WebDriver protocol and handling certain common logic is taken care of already by Appium. This is
+all encoded up as a class which Appium exports for you to use, called `BaseDriver`. It is exported
+from `appium/driver`, so you can use one of these styles to import it and create your *own* class
+that extends it:
+
+```js
+import {BaseDriver} from 'appium/driver';
+// or: const {BaseDriver} = require('appium/driver');
+
+export class MyDriver extends BaseDriver {
+}
+```
+
+### Make your driver available
+
+That's basically it! With a Node.js package exporting a driver class and with a correct Appium
+manifest, you've got yourself an Appium driver! Now it doesn't *do* anything, but you can load it
+up in Appium, start and stop sessions with it, etc...
+
+To make it available to users, you could publish it via NPM. When you do so, your driver will be
+isntallable via the Appium CLI:
+
+```
+appium driver install --source=npm <driver-package-on-npm>
+```
+
+### Developing your driver
```

Comment:
How about installing it from `local` source instead (which is `npm link` under the hood)? This is what I usually do